grammars/css.cson

```diff
@@ -20,294 +20,1237 @@
+        'begin': '(?i)((@)font-face)(?=\\s*|{|/\\*|$)'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.font-face.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '(?!\\G)'
+        'name': 'meta.at-rule.font-face.css'
+        'patterns': [
+          {
+            'include': '#comment-block'
+          }
+          {
+            'include': '#escapes'
+          }
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @media
+        'begin': '(?i)(?=@media(\\s|\\(|/\\*|$))'
+        'end': '}|(?=;)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.end.bracket.curly.css'
+        'patterns': [
+          {
+            'begin': '(?i)(@)media(?=\\s|\\(|/\\*|$)'
+            'beginCaptures':
+              '0':
+                'name': 'keyword.control.at-rule.media.css'
+              '1':
+                'name': 'punctuation.definition.keyword.css'
+            'end': '(?=\\s*[{;])'
+            'name': 'meta.at-rule.media.css'
+            'patterns': [
+              {
+                'include': '#media-query-list'
+              }
+            ]
+          }
+          {
+            'begin': '{'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.section.property-list.begin.bracket.curly.css'
+            'end': '(?=})'
+            'patterns': [
+              {
+                'include': '$self'
+              }
+            ]
+          }
+        ]
+      }
+      {
+        # @page
+        'begin': '(?i)(@)page(?=[\\s:{;]|/\\*|$)'
+        'captures':
+          '0':
+            'name': 'keyword.control.at-rule.page.css'
+          '1':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '(?=\\s*($|[:{;]))'
+        'name': 'meta.at-rule.page.css'
+        'patterns': [
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @counter-style
+        'begin': '(?i)(?=@counter-style([\\s\'"{;]|/\\*|$))'
+        'end': '(?<=})'
+        'patterns': [
+          {
+            'begin': '(?i)\\G(@)counter-style'
+            'beginCaptures':
+              '0':
+                'name': 'keyword.control.at-rule.counter-style.css'
+              '1':
+                'name': 'punctuation.definition.keyword.css'
+            'end': '(?=\\s*{)'
+            'name': 'meta.at-rule.counter-style.header.css'
+            'patterns': [
+              {
+                'include': '#comment-block'
+              }
+              {
+                'include': '#escapes'
+              }
+              {
+                'match': '''(?x)
+                  (?:[-a-zA-Z_]    | [^\\x00-\\x7F])     # First letter
+                  (?:[-a-zA-Z0-9_] | [^\\x00-\\x7F]      # Remainder of identifier
+                    |\\\\(?:[0-9a-fA-F]{1,6}|.)
+                  )*
```

Comment:
Wait, I remember making this lax on purpose, because the pattern for matching numbers is matched first, before `(?:[-a-zA-Z_]    | [^\\x00-\\x7F])` has a chance to match. That just leaves `-`, which won't be highlighted as an identifier.